src/mito2/src/worker/handle_open.rs

```diff
@@ -103,6 +105,9 @@ impl<S: LogStore> RegionWorkerLoop<S> {
                 return;
             }
         };
+        if let Some(wal_entry_receiver) = wal_entry_receiver {
+            opener = opener.wal_entry_reader(Box::new(wal_entry_receiver));
+        };
```

Comment:
Let the opener take an optional reader.
```suggestion
        }
        .wal_entry_reader(wal_entry_receiver.map(|receiver| Box::new(receiver) as _));
```